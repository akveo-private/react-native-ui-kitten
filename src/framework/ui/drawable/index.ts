export {
  Arrow,
  Props as ArrowProps,
} from './arrow/arrow.component';
export {
  CheckMark,
  Props as CheckMarkProps,
} from './checkmark/checkmark.component';
