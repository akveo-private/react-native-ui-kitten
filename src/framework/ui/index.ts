import { styled } from '@kitten/theme';
import {
  Button as ButtonComponent,
  Props as ButtonProps,
} from './button/button.component';
import {
  Text as TextComponent,
  Props as TextProps,
} from './text/text.component';
import {
  Radio as RadioComponent,
  Props as RadioProps,
} from './radio/radio.component';
import {
  RadioGroup as RadioGroupComponent,
  Props as RadioGroupProps,
} from './radioGroup/radioGroup.component';
import {
  Layout as LayoutComponent,
  Props as LayoutProps,
} from './layout/layout.component';
import {
  Toggle as ToggleComponent,
  Props as ToggleProps,
} from './toggle/toggle.component';
import {
  CheckBox as CheckBoxComponent,
  Props as CheckBoxProps,
} from './checkbox/checkbox.component';
import {
  Tab as TabComponent,
  Props as TabProps,
} from './tab/tab.component';
import {
  TabBar as TabBarComponent,
  Props as TabBarProps,
} from './tab/tabBar.component';
import {
  Popover as PopoverComponent,
  Props as PopoverProps,
} from './popover/popover.component';
import {
  Tooltip as TooltipComponent,
  Props as TooltipProps,
} from './tooltip/tooltip.component';
import {
  ViewPager,
  Props as ViewPagerProps,
} from './viewPager/viewPager.component';
import {
  TabView,
  Props as TabViewProps,
} from './tab/tabView.component';
import {
  BottomTabNavigator as BottomTabNavigatorComponent,
  Props as BottomTabNavigatorProps,
} from './tabNavigator/bottomTabNavigator.component';
import {
  TopNavigationBar as TopNavigationBarComponent,
  Props as TopNavigationBarProps,
} from './topNavigationBar/topNavigationBar.component';
import {
  TopNavigationBarAction as TopNavigationBarActionComponent,
  Props as TopNavigationBarActionProps,
} from './topNavigationBar/topNavigationBarAction.component';
import {
  BottomNavigatorTab as BottomNavigatorTabComponent,
  Props as BottomNavigatorTabProps,
} from './tabNavigator/bottomNavigatorTab.component';
import {
  Modal as ModalComponent,
  Props as ModalProps,
} from './modal/modal.component';
import {
  ButtonAlignment,
  ButtonAlignments,
} from './button/type';
import {
  Placement as PopoverPlacement,
  Placements as PopoverPlacements,
} from './popover/type';

const Button = styled<ButtonComponent, ButtonProps>(ButtonComponent);
const Text = styled<TextComponent, TextProps>(TextComponent);
const Radio = styled<RadioComponent, RadioProps>(RadioComponent);
const RadioGroup = styled<RadioGroupComponent, RadioGroupProps>(RadioGroupComponent);
const Layout = styled<LayoutComponent, LayoutProps>(LayoutComponent);
const Toggle = styled<ToggleComponent, ToggleProps>(ToggleComponent);
const CheckBox = styled<CheckBoxComponent, CheckBoxProps>(CheckBoxComponent);
const Tab = styled<TabComponent, TabProps>(TabComponent);
const TabBar = styled<TabBarComponent, TabBarProps>(TabBarComponent);
const Popover = styled<PopoverComponent, PopoverProps>(PopoverComponent);
const Tooltip = styled<TooltipComponent, TooltipProps>(TooltipComponent);
const BottomTabNavigator = styled<BottomTabNavigatorComponent, BottomTabNavigatorProps>(BottomTabNavigatorComponent);
const BottomNavigatorTab = styled<BottomNavigatorTabComponent, BottomNavigatorTabProps>(BottomNavigatorTabComponent);
const TopNavigationBar = styled<TopNavigationBarComponent, TopNavigationBarProps>(TopNavigationBarComponent);
const TopNavigationBarAction =
  styled<TopNavigationBarActionComponent, TopNavigationBarActionProps>(TopNavigationBarActionComponent);
const Modal = styled<ModalComponent, ModalProps>(ModalComponent);

export {
  Button,
  Text,
  Layout,
  LayoutProps,
  Radio,
  RadioProps,
  RadioGroup,
  RadioGroupProps,
  Toggle,
  ToggleProps,
  CheckBox,
  Tab,
  TabBar,
  Popover,
  Tooltip,
  ViewPager,
  TabView,
  ButtonProps,
  CheckBoxProps,
  TabProps,
  TabBarProps,
  PopoverProps,
  TooltipProps,
  ViewPagerProps,
  TabViewProps,
  BottomTabNavigator,
  BottomTabNavigatorProps,
  BottomNavigatorTab,
  BottomNavigatorTabProps,
  TopNavigationBar,
  TopNavigationBarProps,
  TopNavigationBarAction,
  TopNavigationBarActionProps,
  Modal,
  ModalProps,
  ButtonAlignment,
  ButtonAlignments,
  PopoverPlacement,
  PopoverPlacements,
};
