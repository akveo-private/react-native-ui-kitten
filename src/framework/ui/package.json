{
  "name": "@kitten/ui",
  "version": "0.0.1",
  "description": "kitten ui components",
  "repository": "git+https://github.com/akveo/react-native-ui-kitten.git",
  "author": "akveo <contact@akveo.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/akveo/react-native-ui-kitten/issues"
  },
  "homepage": "https://github.com/akveo/react-native-ui-kitten#readme",
  "peerDependencies": {
    "@kitten/theme": "file:../theme"
  }
}
