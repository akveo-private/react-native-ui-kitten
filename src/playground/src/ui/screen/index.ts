export { HomeScreen } from './home.component';
export { RadioScreen } from './radio.component';
