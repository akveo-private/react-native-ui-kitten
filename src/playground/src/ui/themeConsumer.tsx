import React from 'react';
import {
  View,
  ViewProps,
  StyleSheet,
  ImageProps,
  Image,
} from 'react-native';
import { Button } from '@kitten/ui';
import { ThemeContext } from './themeContext';
import { StyleType, ThemedComponentProps, ThemeType, withStyles } from '@kitten/theme';

interface Props extends ThemedComponentProps {
  children: React.ReactElement<any>;
}

export class ThemeConsumerComponent extends React.Component<Props> {

  private isThemeButtonDisabled = (currentTheme: string, buttonResponsibility: string): boolean => {
    if (currentTheme === 'light' && buttonResponsibility === 'light') {
      return true;
    } else if (currentTheme === 'dark' && buttonResponsibility === 'light') {
      return false;
    } else if (currentTheme === 'light' && buttonResponsibility === 'dark') {
      return false;
    } else if (currentTheme === 'dark' && buttonResponsibility === 'dark') {
      return true;
    }
  };

  private renderThemeButtonIcon = (style: StyleType,
                                   disabled: boolean): React.ReactElement<ImageProps> => {

    const iconUri: string = disabled ?
      'https://akveo.github.io/eva-icons/fill/png/128/checkmark.png' :
      'https://akveo.github.io/eva-icons/fill/png/128/plus.png';

    return (
      <Image
        style={style}
        source={{ uri: iconUri }}
      />
    );
  };

  private renderContent = ({ currentTheme, toggleTheme }): React.ReactElement<ViewProps> => {
    const lightBtnDis: boolean = this.isThemeButtonDisabled(currentTheme, 'light');
    const darkBtnDis: boolean = this.isThemeButtonDisabled(currentTheme, 'dark');

    return (
      <View style={this.props.themedStyle.container}>
        <View style={this.props.themedStyle.themeButtonsContainer}>
          <Button
            disabled={lightBtnDis}
            icon={(style: StyleType) => this.renderThemeButtonIcon(style, lightBtnDis)}
            onPress={() => toggleTheme('light')}>
            Light
          </Button>
          <Button
            disabled={darkBtnDis}
            icon={(style: StyleType) => this.renderThemeButtonIcon(style, darkBtnDis)}
            onPress={() => toggleTheme('dark')}>
            Dark
          </Button>
        </View>
        {this.props.children}
      </View>
    );
  };

  public render(): React.ReactNode {
    return (
      <ThemeContext.Consumer>
        {this.renderContent}
      </ThemeContext.Consumer>
    );
  }
}

export const ThemeConsumer = withStyles(ThemeConsumerComponent, (theme: ThemeType) => ({
  container: {
    flex: 1,
  },
  themeButtonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    paddingVertical: 16,
    backgroundColor: theme['background-color-default-2'],
  },
}));
