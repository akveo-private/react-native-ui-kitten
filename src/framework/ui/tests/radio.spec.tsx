import React from 'react';
import {
  render,
  fireEvent,
} from 'react-native-testing-library';
import {
  StyleProvider,
  VARIANT_DEFAULT,
} from '@rk-kit/theme';
import {
  Mappings as mappings,
  Theme as theme,
} from './config';
import { Radio, RadioProps } from '@rk-kit/ui';

const componentTestId = '@radio/component';

class Mock extends React.Component<RadioProps> {

  static defaultProps: RadioProps = {
    checked: false,
    disabled: false,
    variant: VARIANT_DEFAULT,
  };

  render() {
    return (
      <StyleProvider mapping={mappings} theme={theme}>
        <Radio
          testID={componentTestId}
          {...this.props}
        />
      </StyleProvider>
    );
  }
}

describe('@radio: component checks', () => {

  it('default variant renders properly', async () => {
    const component = render(
      <Mock />,
    );
    const radioComponent = component.getByTestId(componentTestId);
    expect(radioComponent).not.toBeNull();
    expect(radioComponent).not.toBeUndefined();
  });

  it('emits onChange', async () => {
    const onChange = jest.fn();
    const component = render(
      <Mock onChange={onChange}/>,
    );
    const radioTouchable = component.getByTestId('@radio/touchable');
    fireEvent.press(radioTouchable);

    expect(onChange).toBeCalled();
  });

});
