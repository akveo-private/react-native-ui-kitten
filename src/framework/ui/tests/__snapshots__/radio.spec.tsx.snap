// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@radio: matches snapshot active 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#A6AEBD",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot active checked 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "#3366FF",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot active checked: checked 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "#3366FF",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot active: default 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#A6AEBD",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot checked 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "#3366FF",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot checked disabled 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  disabled={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#DDE1EB",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "#A6AEBD",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot default 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#A6AEBD",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot disabled 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={false}
  disabled={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  requestStateStyle={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#A6AEBD",
      "borderWidth": 2,
      "highlightColor": "transparent",
      "highlightSize": 60,
      "innerSize": 24,
      "selectColor": "transparent",
      "size": 36,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 60,
            "opacity": 0,
            "width": 60,
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#DDE1EB",
            "borderRadius": 18,
            "borderWidth": 2,
            "height": 36,
            "width": 36,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;
